src/components/custom/Button.tsx

```
@@ -0,0 +1,60 @@
+    const baseStyles = "inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50";
+    
+    const combinedClassName = cn(
+      baseStyles,
+      variantStyles[variant],
+      sizeStyles[size],
+      className
+    );
+
+    if (asChild) {
+      return (
+        <span className={combinedClassName} ref={ref as any} {...props}>
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)